test/cache.ts

```diff
@@ -127,26 +99,39 @@ test('Cached response should have got options', async t => {
 	t.is(secondResponse.request.gotOptions.auth, options.auth);
 });
 
-test('Cache error throws got.CacheError', async t => {
-	const endpoint = '/no-store';
+test('cache error throws `got.CacheError`', withServer, async (t, server, got) => {
+	server.get('/', (_request, response) => {
+		response.end('ok');
+	});
+
 	const cache = {};
 
-	const error = await t.throwsAsync(got(`${s.url}${endpoint}`, {cache}));
-	t.is(error.name, 'CacheError');
+	await t.throwsAsync(() => got({cache}), got.CacheError);
 });
 
-test('doesn\'t cache response when received HTTP error', async t => {
-	const endpoint = '/first-error';
+test('doesn\'t cache response when received HTTP error', withServer, async (t, server, got) => {
+	let calledFirstError = false;
+	server.get('/', (_request, response) => {
+		if (calledFirstError) {
+			response.end('ok');
+			return;
+		}
+
+		calledFirstError = true;
+		response.statusCode = 502;
+		response.end('received 502');
+	});
+
 	const cache = new Map();
 
-	const {statusCode, body} = await got(`${s.url}${endpoint}`, {cache, throwHttpErrors: false});
+	const {statusCode, body} = await got({cache, throwHttpErrors: false});
 	t.is(statusCode, 200);
 	t.deepEqual(body, 'ok');
 });
 
-test('DNS cache works', async t => {
+test('DNS cache works', withServer, async (t, _server, got) => {
 	const map = new Map();
-	await t.notThrowsAsync(got('https://example.com', {dnsCache: map}));
+	await t.notThrowsAsync(() => got('https://example.com', {dnsCache: map}));
```

Comment:
While AVA does support this, there's no point in doing it this way, as you can just pass the promise directly, like it did previously. Applies to all the places you changed this.

Reply:
But if there's something wrong then `got()` could throw. But the test would fail anyway I think...

Reply:
`got()` should never throw. If it throws, it's a bug we should fix. It should **always** reject the promise. Promise-returning functions should never throw. `got.stream()` however, can throw.

Reply:
Okay, doing that now :D